test/e2e/specs/site-editor/font-library.spec.js

```diff
@@ -15,15 +15,64 @@ test.describe( 'Font Library', () => {
 		} );
 
 		test( 'should display the "Manage Fonts" icon', async ( { page } ) => {
-			await page.getByRole( 'button', { name: /styles/i } ).click();
+			await page.getByRole( 'button', { name: 'Styles' } ).click();
 			await page
-				.getByRole( 'button', { name: /typography styles/i } )
+				.getByRole( 'button', { name: 'Typography Styles' } )
 				.click();
 			const manageFontsIcon = page.getByRole( 'button', {
-				name: /manage fonts/i,
+				name: 'Manage Fonts',
 			} );
 			await expect( manageFontsIcon ).toBeVisible();
 		} );
+
+		test( 'should allow user to upload a local font', async ( {
+			page,
+		} ) => {
+			await page.getByRole( 'button', { name: 'Styles' } ).click();
+			await page
+				.getByRole( 'button', { name: 'Typography Styles' } )
+				.click();
+			await page
+				.getByRole( 'button', {
+					name: 'Manage Fonts',
+				} )
+				.click();
+
+			// Delete test font family (Exo 2) if it exists
+			await expect(
+				page.getByRole( 'tab', { name: 'Upload' } )
+			).toBeVisible( { timeout: 80000 } );
+
+			if (
+				await page.getByRole( 'button', { name: 'Exo 2' } ).isVisible()
+			) {
+				await page.getByRole( 'button', { name: 'Exo 2' } ).click();
+				await page.getByRole( 'button', { name: 'Delete' } ).click();
+				await page.getByRole( 'button', { name: 'Delete' } ).click();
+				await page.getByRole( 'tab', { name: 'Library' } ).click();
+			}
+
+			// Upload local fonts
+			await page.getByRole( 'tab', { name: 'Upload' } ).click();
+			const fileChooserPromise = page.waitForEvent( 'filechooser' );
+			await page.getByRole( 'button', { name: 'Upload Font' } ).click();
+			const fileChooser = await fileChooserPromise;
+			await fileChooser.setFiles( [
+				'./test/e2e/assets/Exo2-Regular.woff',
+				'./test/e2e/assets/Exo2-SemiBoldItalic.woff2',
+			] );
+
+			// Check fonts were installed
+			await expect(
+				page
+					.getByLabel( 'Upload' )
+					.getByText( 'Fonts were installed successfully.' )
+			).toBeVisible( { timeout: 40000 } );
+			await page.getByRole( 'tab', { name: 'Library' } ).click();
+			await expect(
+				page.getByRole( 'button', { name: 'Exo 2' } )
+			).toBeVisible();
```

Comment:
```suggestion
			// Provides coverage for https://github.com/WordPress/gutenberg/issues/60040.
			await expect(
				page.getByRole( 'button', { name: 'Exo 2' } )
			).toBeVisible();
```

Ideally we'd also check the CSS is created.